mail_template_multi_attachment/models/mail_template.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import base64
+
+from odoo import _, exceptions, fields, models
+from odoo.tools.safe_eval import safe_eval, time
+
+
+class MailTemplate(models.Model):
+    _inherit = "mail.template"
+
+    template_report_ids = fields.One2many(
+        comodel_name="mail.template.report",
+        inverse_name="mail_template_id",
+    )
+
+    # pylint: disable=redefined-outer-name
+    def generate_email(self, res_ids, fields=None):
+        """
+        Inherit to generate attachments.
+        Inspired from original mail.template,generate_email(...) from Odoo.
+        :param res_ids: int or list of int
+        :param fields:
+        :return: dict
+        """
+        self.ensure_one()
+        multi_mode = True
+        results = super(MailTemplate, self).generate_email(res_ids, fields=fields)
+        if not self.template_report_ids:
+            return results
+        if isinstance(res_ids, int):
+            multi_mode = False
+            results = {res_ids: results}
+        # Generate attachments (inspired from Odoo); Just add new attachments
+        # into 'attachments' key
+        for res_id, values in results.items():
+            attachments = values.setdefault("attachments", [])
+            for template_report in self.template_report_ids:
+                report = template_report.report_template_id
+                print_report_name = (
+                    template_report.report_name or report.print_report_name
+                )
+                report_name = False
+                if print_report_name:
+                    report_name = safe_eval(
+                        print_report_name,
+                        {"object": self.env[self.model].browse(res_id), "time": time},
+                    )
+                report_service = report.report_name
+
+                if report.report_type in ["qweb-html", "qweb-pdf"]:
+                    result, report_format = report._render_qweb_pdf(
+                        report, res_ids=[res_id]
+                    )
+                else:
+                    res = report.render([res_id])
```

Comment:
Since 14.0, this method does not exist anymore. Replaced with _render (https://github.com/odoo/odoo/blob/32d159921443daa3cf559bcc2cd605094e34b6ff/odoo/addons/base/models/ir_actions_report.py#L922)

Reply:
Good point. And in fact it has no test coverage :wink:

Reply:
changes done, added new test case @simahawk